blogposts/2021/how-we-built-batch-changes.md

```diff
@@ -0,0 +1,202 @@
+---
+title: 'How we built Batch Changes, explained in 5 pull requests'
```

Comment:
or just "How we avoided the biggest dangers of iteration-base product development"

Reply:
I really think we should keep the "5 pull requests" in there somehow, because that's what would make me click. "iteration-based product development" in a header is something I would skip. But "you build something in 5 PRs? which ones?" is what would make me click & read.

Reply:
Agreed, I was immediately curious: a story told in 5 PRs? Which ones?

Reply:
What about: 

"How we avoided the biggest dangers of iteration-based product development explained in 5 pull requests"

Reply:
Shorter version: "Avoiding the pitfalls of iteration-based development explained in 5 pull requests"

Reply:
Hm, I think it depends on the audience we're targeting here. "How we built X" is always interesting to me, especially on a tech company blog. Stuff that sounds "product development-y" is hit and miss to me.

My original idea was:

```
How we built Batch Changes - in 5 pull requests
```

Reply:
My concern with leading with Batch Changes is that many people have never heard of it. Our audience for this post is software engineers, but not necessarily currents users, or even someone who has heard of Sourcegraph before. 

My question for you: is "how we built X" is universally interesting, regardless of product and company familiarity?

Reply:
> My question for you: is "how we built X" is universally interesting, regardless of product and company familiarity?

I wouldn't say regardless of company familiarity. When I know that the company builds interesting stuff I will definitely be more likely to read.

But given two options:

1. "How we built X" with X being a thing that sounds like a product name I don't know about
2. more generic sounding title

and the same company familiarity, I'd still prefer 1 more.

Reply:
I love the "in 5 pull requests" idea. This immediately hooks me to the article. Wait. What? Oh, nice. What I like from a more analytical perspective is that this title doesn't sound like other titles I'm used too. And I want to know why!! So I keep on reading.